### DISCUSSION TITLE
can't reproduce HNN-GUI model in hnn_core

### DISCUSSION THREAD

**Discussion by EPinzuti** [jonescompneurolab/hnn-core#693] [Q&A]
Hi,

I simulated a model in the hnn-gui  changing the function L5_pyramidal.py to have the calcium model. After the simulation I saved the parameters and imported them in the hnn-core. Using the calcium_model(), I was thinking to see similar results, but on average the shape of the simulation does not match the one I get in the hnn-gui. Are there other parameters that need to be changed ?

thanks a lot 
Edoardo

**Comment by rythorpe** [comment]
Do you mean that you changed the [L5Pyr.__biophys_dends](https://github.com/jonescompneurolab/hnn/blob/a09bb082b2c712f025faa9d755f3edf50f23d36a/L5_pyramidal.py#L323-L375) method in `L5_pyramidal.py` of the [legacy] hnn repository? If so, could you provide the exact code you used to change the function so that we can reproduce the issue?

Also, were you trying to reproduce the calcium model from Kohl et al. 2021, or just wanting to implement a model with more realistic dendritic calcium dynamics? Note that the model used in Kohl et al. 2021 is actually different from [calcium_model](https://github.com/jonescompneurolab/hnn-core/blob/1216038b9cb8a753fa4162e77241f0b860e84978/hnn_core/network_models.py#L262) as defined in hnn-core. We just discovered this discrepancy, but are encouraging everyong to use the version distributed with hnn-core because it is tuned to be more realistic. 

Copying @darcywaller because she has also been using both the old GUI and hnn-core to run simulations on the updated calcium model and might be able to give us insight with this issue moving forward.

**Comment by EPinzuti** [comment]
Thanks a lot for your help.

Yes, I uploaded the L5_pyramidal.py provided by Kohl  with the changed _biophys_dends method.  I did not change the code myself.  Running now the model with default parameters produced a simulation which is not equivalent to the calcium_model() and default parameters of the hnn-core, as you pointed  they are different. I wanted to use the hnn-GUI to tune the parameters and then import them in the hnn-core. I was using the GUI to find optimal parameter for my data using a more realistic dendritic calcium dynamic. I guess  I should only work with the hnn-core then.

**Comment by jasmainak** [comment]
hey @EPinzuti hnn-core has optimization routines. Why not use them to [tune the parameters](https://jonescompneurolab.github.io/hnn-core/stable/generated/hnn_core.optimization.optimize_evoked.html#hnn_core.optimization.optimize_evoked)? Backwards compatibility is going to become more and more tricky as time passes by ... 

I remember having gotten similar results using the `calcium_model` as the Kohl model (even though there are slight differences).

**Reply by EPinzuti** [reply]
sorry I also saw some minor differences I did not mean that they are completely different. Yes I will use only the hnn-core I just wanted to be sure that I was not doing  something wrong.

**Comment by jasmainak** [comment]
We should certainly document the differences between the Kohl et al model and the calcium model ... cc @darcywaller

**Comment by darcywaller** [comment]
As alluded to, we are aware of the fact that the calcium_model() in hnn-core produces different results from using the L5_pyr Kohl et al. file mod to the GUI. I am currently working on getting a function set up for the Kohl model in HNN core so that this is less confusing. 

However, it is worth noting that the calcium_model() is configured to replicate the network that results from the L5_Pyr GUI mod _when you use the .param files from the Kohl paper._ Simply adding the modified L5_pyr file in the GUI will not guarantee you accomplish this because the Kohl et al. 2021 model has some network (non-drive) parameters that are not the HNN somatosensory defaults, which are built into network object construction in hnn-core but not in the GUI. There are certain ranges of parameters (as you've probably noticed) at which this becomes more obvious if you try to compare "calcium GUI" versus the calcium_model() in core or the unmodified GUI without ensuring that the network connectivity is exactly the same.

**Reply by darcywaller** [reply]
So, in my understanding, in essence this is less of a concern about how the Kohl model and calcium model differ and more of a use case for the modified GUI in which caution is needed. I think the hard-coded construction of those aspects of the model in hnn-core prevents accidental misuse better than in the case of the single mod file for the GUI.
